addons/stock/models/stock_inventory.py

```diff
@@ -164,6 +170,46 @@ def action_reset_product_qty(self):
         return True
 
     def action_validate(self):
+        # Check the serial numbers uniquiness
+        serials = self.line_ids.filtered(lambda l: l.product_id.tracking == 'serial' and l.prod_lot_id and l.theoretical_qty < l.product_qty)
+        wiz_lines = []
+        for line in serials:
+            similar_quant = self.env['stock.quant'].search([
+                ('lot_id', '=', line.prod_lot_id.id),
+                ('location_id', '!=', line.location_id.id),
+                '|',
+                ('location_id.usage', 'not in', ('supplier', 'inventory', 'production')),
+                ('location_id.scrap_location', '=', True),
+                ('quantity', '>', 0)
+            ])
+            # Extract the quant to correct
+            if len(similar_quant) > 1:
+                same_usage = similar_quant.filtered(lambda q: q.location_id.usage == line.location_id.usage)
+                if same_usage:
+                    similar_quant = same_usage[0]
+                else:
+                    similar_quant = similar_quant[0]
+            # Fill the wizard lines
+            if similar_quant and similar_quant.quantity - sum(self.line_ids.filtered(lambda line: line.correction_line == 'warning' and line.prod_lot_id == similar_quant.lot_id).mapped('theoretical_qty')) != 0:
```

Comment:
I would remove the line.correction_line == 'warning' if the user did a manual correction?

Reply:
.mapped('product_qty') ?